protocols/imap/src/main/java/org/apache/james/imap/decode/parser/GetAnnotationCommandParser.java

```diff
@@ -29,119 +28,99 @@
 import org.apache.james.imap.api.display.HumanReadableText;
 import org.apache.james.imap.api.process.ImapSession;
 import org.apache.james.imap.decode.ImapRequestLineReader;
-import org.apache.james.imap.decode.ImapRequestLineReader.CharacterValidator;
 import org.apache.james.imap.decode.base.AbstractImapCommandParser;
 import org.apache.james.imap.message.request.GetAnnotationRequest;
 import org.apache.james.imap.message.request.GetAnnotationRequest.Depth;
 import org.apache.james.protocols.imap.DecodingException;
 
 import java.util.Set;
-import java.util.regex.Pattern;
 
 public class GetAnnotationCommandParser extends AbstractImapCommandParser {
     private static final CharMatcher ENDOFLINE_PATTERN = CharMatcher.isNot('\n').and(CharMatcher.isNot('\r'));
-    private final static byte[] MAXSIZE = "MAXSIZE".getBytes();
-    private final static byte[] DEPTH = "DEPTH".getBytes();
+    private final static String MAXSIZE = "MAXSIZE";
+    private final static String DEPTH = "DEPTH";
 
     public GetAnnotationCommandParser() {
         super(ImapCommand.authenticatedStateCommand(ImapConstants.GETANNOTATION_COMMAND_NAME));
     }
 
     @Override
-    protected ImapMessage decode(ImapCommand command, ImapRequestLineReader request, String tag, ImapSession session)
+    protected ImapMessage decode(ImapCommand command, ImapRequestLineReader requestReader, String tag, ImapSession session)
             throws DecodingException {
+        GetAnnotationRequest.GetAnnotationRequestBuilder builder = new GetAnnotationRequest.GetAnnotationRequestBuilder(tag, command);
         try {
-            GetAnnotationRequest annotationRequest = new GetAnnotationRequest(tag, command, request.mailbox());
+            builder.mailboxName(requestReader.mailbox());
```

Comment:
Move the builder creation in the try catch.
Extract the inside of the try block to a new method.